modules/ballerina-core/src/main/java/org/ballerinalang/model/types/TypeLattice.java

```diff
@@ -105,58 +107,121 @@ public static void loadExplicitCastLattice(SymbolScope scope) {
         explicitCastLattice.addVertex(anyV, false);
         explicitCastLattice.addVertex(connectorV, false);
 
-        explicitCastLattice.addEdge(intV, floatV, NativeCastMapper.INT_TO_FLOAT_FUNC);
-        explicitCastLattice.addEdge(intV, stringV, NativeCastMapper.INT_TO_STRING_FUNC);
-        explicitCastLattice.addEdge(intV, booleanV, NativeCastMapper.INT_TO_BOOLEAN_FUNC);
-        explicitCastLattice.addEdge(intV, intV, NativeCastMapper.INT_TO_INT_FUNC);
-        explicitCastLattice.addEdge(intV, anyV, NativeCastMapper.INT_TO_ANY_FUNC);
-        explicitCastLattice.addEdge(intV, jsonV, NativeCastMapper.INT_TO_JSON_FUNC);
-        
-        explicitCastLattice.addEdge(floatV, floatV, NativeCastMapper.FLOAT_TO_FLOAT_FUNC);
-        explicitCastLattice.addEdge(floatV, stringV, NativeCastMapper.FLOAT_TO_STRING_FUNC);
-        explicitCastLattice.addEdge(floatV, booleanV, NativeCastMapper.FLOAT_TO_BOOLEAN_FUNC);
-        explicitCastLattice.addEdge(floatV, intV, NativeCastMapper.FLOAT_TO_INT_FUNC);
-        explicitCastLattice.addEdge(floatV, anyV, NativeCastMapper.FLOAT_TO_ANY_FUNC);
-        explicitCastLattice.addEdge(floatV, jsonV, NativeCastMapper.FLOAT_TO_JSON_FUNC);
-        
-        explicitCastLattice.addEdge(stringV, floatV, NativeCastMapper.STRING_TO_FLOAT_FUNC);
-        explicitCastLattice.addEdge(stringV, stringV, NativeCastMapper.STRING_TO_STRING_FUNC);
-        explicitCastLattice.addEdge(stringV, intV, NativeCastMapper.STRING_TO_INT_FUNC);
-        explicitCastLattice.addEdge(stringV, anyV, NativeCastMapper.STRING_TO_ANY_FUNC);
-        explicitCastLattice.addEdge(stringV, jsonV, NativeCastMapper.STRING_TO_JSON_FUNC);
-        
-        explicitCastLattice.addEdge(booleanV, stringV, NativeCastMapper.BOOLEAN_TO_STRING_FUNC);
-        explicitCastLattice.addEdge(booleanV, booleanV, NativeCastMapper.BOOLEAN_TO_BOOLEAN_FUNC);
-        explicitCastLattice.addEdge(booleanV, intV, NativeCastMapper.BOOLEAN_TO_INT_FUNC);
-        explicitCastLattice.addEdge(booleanV, floatV, NativeCastMapper.BOOLEAN_TO_FLOAT_FUNC);
-        explicitCastLattice.addEdge(booleanV, anyV, NativeCastMapper.BOOLEAN_TO_ANY_FUNC);
-        explicitCastLattice.addEdge(booleanV, jsonV, NativeCastMapper.BOOLEAN_TO_JSON_FUNC);
+        for (boolean returnErrors: errorReturnStates) {
```

Comment:
Certain casting operations are allowed only for the multi-return casting operation.
eg:  in the above sample `newEmp, err = (Employee) person` is valid and `newEmp = (Employee) person` invalid.

Therefore, we need to maintain mapping functions for both scenarios separately. I have kept a single lattice to hold both mapping functions, differentiated by a flag ('returnErrors' - which indicates whether this edge is for multi-return or single return), rather than maintaining two different lattices.

This loop adds the mapping functions that are supported for both scenarios.